mantidproject/mantid
# Use Native Enums In ISIS SANS

**Description of work.**
Converts all of the SANS scripts to use native Python enums, this:
- Improves readability / debugging 
- Removes custom enum handling we currently do
- ~~Improves performance~~ Only by 1%
- Clears some edge cases around the serializer

**Note to the reviewer(s):** The only significant code changes are in the commits which do not follow the "Convert x to Py Enum" pattern. All other commits are done with automatic tooling

**To test:**
- Run some data through the GUI and click the various tabs
- Check that tests build


Fixes #25358.

Does not require release notes as it's a long standing maintenance job

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
